request_router.cc

```diff
@@ -1,103 +1,63 @@
-#include "request_router.h"
 #include "request_handlers.h"
+#include "request_router.h"
 
-#include <boost/tokenizer.hpp>
-
-// TODO: Split out HTTP parsing into its own class
-bool RequestRouter::routeRequest(const ServerConfig& server_config,
-                                 const char* request_buffer,
-                                 const size_t& request_buffer_size,
-                                 char* &response_buffer,
-                                 size_t& response_buffer_size) {
+typedef std::map<const std::string, const std::unique_ptr<RequestHandler>> HandlersMap;
 
-  // Tokenize the request so it's easy to extract the resource path
-  const std::string request(request_buffer);
-  boost::char_separator<char> separator(" ");
-  boost::tokenizer<boost::char_separator<char>> tokens(request, separator);
+bool RequestRouter::BuildRoutes(const ServerConfig& server_config) {
+  // TODO: Build uri_prefix -> RequestHandler* map
+  // Iterate over uri_prefix -> handler_name map
+  for (const auto path : server_config.allPaths()) {
+    const std::string uri_prefix = path.first;
```

Comment:
I think the syntax for accessing the elements of the `unordered_map` will be `path->first` and `path->second`